src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,77 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import { FilterType } from '../../types/FilterType';
+
+export type Props = {
+  onDeleteCompletedTodos(): void,
+  todosLeft: number,
+  setFilterType(type: FilterType): void,
+  filterType: FilterType,
+  completedTodosLength: number,
+};
+
+export const Footer: React.FC<Props> = ({
+  onDeleteCompletedTodos,
+  todosLeft,
+  setFilterType,
+  filterType,
+  completedTodosLength,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${todosLeft} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={classNames(
+            'filter__link', { selected: FilterType.All === filterType },
+          )}
+          onClick={() => setFilterType(FilterType.All)}
+        >
```

Comment:
Do not use anonymous function. Create seperate handler for it. Check other place too